ravenframework/TSA/Fourier.py

```diff
@@ -192,24 +246,6 @@ def getParamsAsVars(self, params):
           rlz[f'{baseWave}__{stat}'] = value
     return rlz
 
-
-  def generate(self, params, pivot, settings):
```

Comment:
With Fourier being implemented as a TimeSeriesTransformer instead of a TimeSeriesGenerator, I wanted to avoid using the same `generate()` method name used in TimeSeriesGenerator. If we revert this back to being a generator (and relaxing the restriction on generators being stochastic), I'll revert the name change.

Reply:
Okay, help me understand that, and maybe talking this over in a meeting makes more sense. Fourier projects the signal from signal space to a Hilbert space, but the result is not a transformation of the data, but a characterization of the data, as it seems to me. Help me see how you're seeing Fourier as a transformer.

Reply:
I see it as a transformer in the sense that you give it a signal and it returns a modified signal. It certainly characterizes as well, but I see the characterization of the signal as separate thing from the operation of modifying the signal based on the results of that characterization. I'd certainly be open to discussing this and some of the broader design decisions in a meeting instead of through a Github PR review..